shared/middleware/files_api.rb

```diff
@@ -92,6 +79,10 @@ def allowed_file_type?(endpoint, extension)
     # read the entire request before considering rejecting it, otherwise varnish
     # may return a 503 instead of whatever status code we specify.
     body = request.body.read
+
+    owner_id, _ = storage_decrypt_channel_id(encrypted_channel_id)
+    not_authorized unless owner_id == storage_id('user')
```

Comment:
StorageApps is analogous to AssetBucket / SourceBucket / BucketHelper so I assume you're asking whether we should set @storage_id in the initializer of BucketHelper. Is that correct? It turns out that we don't need to access the storage_id within BucketHelper so there is no reason to do this.

FilesApi gets passed as a class into the configuration layers for Rails and Sinatra, and may be initialized outside of any request context, so we can't set @storage_id in the initializer of FilesApi either.

Generally, the strategy we discussed is to move HTTP status codes and access to cookies out of the helpers and into the _api.rb files, so over time hopefully we move the @storage_id defs out of StorageApps and the other helpers too.